cookbook/src/app/file-upload/file-upload.component.html

```diff
@@ -1,7 +1,7 @@
 <input type="file" class="file-input" (change)="onFileSelected($event)" #fileUpload>
 
 <div class="file-upload">
-  <button (click)="fileUpload.click()">
+  <button type="button" (click)="fileUpload.click()">
```

Comment:
This is neccessary, because otherwise the default behaiviour of a button in a form is to submit the form.